### PR TITLE
Add basic developer quick start

### PR DESCRIPTION
**Pull request by jamiefolsom**
- Assuming the audience is not just internal, but also wider edx community, I included some basic parameters regarding our github workflow.
- Any thoughts on formatting are welcome.
- Devstack setup cribbed from @carsongee's [slides](http://mitodl.github.io/edx-dev-intro-slides/#6).

### CODE REVIEW

**Comment by pdpinch** [comment]
@pwilkins can you review this?

**Comment by pwilkins** [comment]
@jamiefolsom Have you run this through Sphinx for error checking?

**Comment by pwilkins** [comment]
@jamiefolsom Have you run this through Sphinx for error checking?

**Comment by pwilkins** [comment]
Oops, sorry.  This doesn't have a Sphinx project so you can't easily do that.

**Comment by ShawnMilo** [comment]
There's one error (just running `rst2pdf`):

```bash
 $ rst2pdf  DEVELOPMENT.rst 
DEVELOPMENT.rst:5: (WARNING/2) Title underline too short.

Setup (including devstack setup)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by jamiefolsom** [comment]
@ShawnMilo good catch. fixed.

**Comment by pwilkins** [comment]
One thing to change: We have followed a convention in our doc of documenting command line instructions in a way that allows developers to cut and paste them into the console.  In this case that would require you to remove the numbering.

File: DEVELOPMENT.rst [hunk 1]
```diff
@@ -0,0 +1,39 @@
+Developing on edx-sga
+==============================
+
+Setup (including devstack setup)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Install vagrant: http://docs.vagrantup.com/v2/installation/
+2. Install virtualbox: https://www.virtualbox.org/wiki/Downloads
+3. ``mkdir devstack``
```
Comment (pwilkins) [inline DEVELOPMENT.rst]: You can bracket your command lines with:
```bash
<your commands>
```

File: DEVELOPMENT.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+13. ``cd /edx/app/edxapp/themes/``
+14. ``pip uninstall edx-sga`` (since it's part of the edx distribution, we have to remove the installed version)
+15. ``cd edx-sga/``
+16. ``pip install -e .``
+17. ``paver run_all_servers``
+
+You should now see your fork of the most recent master branch of edx-sga running in the LMS.
+
+Developing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (pwilkins) [inline DEVELOPMENT.rst]: nit: strict .RST rules require that the formatting line is the same length as the text it formats.

File: DEVELOPMENT.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
+Developing on edx-sga
+==============================
```
Comment (pwilkins) [inline DEVELOPMENT.rst]: nit: strict .RST rules require that the formatting line is the same length as the text it formats.

**Comment by jamiefolsom** [comment]
@pwilkins sure; done.

File: DEVELOPMENT.rst [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+#. Write Code, then::
+
+    git add .    
+    git commit -m "Description of feature added."    
+    git push origin feature/your-name/name-of-feature    
+
+#. Rebase your branch against mitodl/master and resolve any conflicts, following this process_.
+#. Open a pull request from your fork/feature branch to mitodl/master
+
+Also, see testing_. Javascript testing will be added soon.
```
Comment (pwilkins) [inline DEVELOPMENT.rst]: Make the URL visible (an edX doc convention).  Remove the promise of Javascript testing.  We don't need to assert our good intentions, we just have to deliver.  :-)

File: DEVELOPMENT.rst [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    vagrant up
+    cd themes/
+
+#. Fork https://github.com/mitodl/edx-sga.git to your own github account.
+#. Set up your development environment::
+
+    git clone https://github.com/your-name/edx-sga.git    
+    vagrant ssh    
+    sudo su edxapp    
+    cd /edx/app/edxapp/themes/
```
Comment (pwilkins) [inline DEVELOPMENT.rst]: If you have sudo'ed to become the ``edxapp`` user, the themes directory is ``./themes``.

File: DEVELOPMENT.rst [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    paver run_all_servers    
+
+You should now see your fork of the most recent master branch of edx-sga running in the LMS.
+
+Developing
+~~~~~~~~~~
+
+#. In your host filesystem::
+
+    cd /path/to/devstack/themes/edx-sga
```
Comment (pwilkins) [inline DEVELOPMENT.rst]: consider changing this to ``/path/to/devstack/edx-platform/themes/edx-sga``.

File: DEVELOPMENT.rst [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+#. `Install vagrant <http://docs.vagrantup.com/v2/installation/>`_.
+#. `Install virtualbox <https://www.virtualbox.org/wiki/Downloads>`_.
+#. Set up devstack::
+
+    mkdir devstack
+    cd devstack
+    curl -L https://raw.githubusercontent.com/edx/configuration/master/vagrant/release/devstack/Vagrantfile > Vagrantfile
+    vagrant plugin install vagrant-vbguest
+    vagrant up
+    cd themes/
```
Comment (pwilkins) [inline DEVELOPMENT.rst]: Consider moving this step into the next section since it's a prereq for those steps.

**Comment by jamiefolsom** [comment]
Did those things. Regarding URLs: should they all be visible? Thanks!

**Comment by pwilkins** [comment]
Yes, all should be visible.  Sorry if I missed some, like Vagrant and VirtualBox.

**Comment by jamiefolsom** [comment]
Thanks, @pwilkins!

**Comment by jamiefolsom** [comment]
Oops, didn't get to squash commits.

**Pull request merged**